producer-c/producer-cloudwatch-integ/jobs/runner.groovy

```diff
@@ -0,0 +1,216 @@
+import jenkins.model.*
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
+
+START_TIMESTAMP = new Date().getTime()
+HAS_ERROR = false
+
+RUNNING_NODES=0
+
+CREDENTIALS = [
+    [
+        $class: 'AmazonWebServicesCredentialsBinding', 
+        accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+        credentialsId: 'SDK_CANARY_CREDS',
+        secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+    ]
+]
+
+def buildProducer() {
+  sh  """
+    cd $WORKSPACE/producer-c/producer-cloudwatch-integ && 
+    mkdir -p build
+    cd build && 
+    sudo cmake .. && 
+    sudo make 
+  """
+}
+
+def buildConsumer(envs) {
+  withEnv(envs) {
+      sh '''
+        PATH="$JAVA_HOME/bin:$PATH"
+        export PATH="$M2_HOME/bin:$PATH"
+        cd $WORKSPACE/consumer-java/aws-kinesis-video-producer-sdk-canary-consumer
+        mvn package
+        touch tmp_jar
+        mvn -Dmdep.outputFile=tmp_jar dependency:build-classpath
+      '''
+  }
+}
+  
+def withRunnerWrapper(envs, fn) {
+    withEnv(envs) {
+        withCredentials(CREDENTIALS) {
+            try {
+                fn()
+            } catch (FlowInterruptedException err) {
+                echo 'Aborted due to cancellation'
+                throw err
+            } catch (err) {
+                HAS_ERROR = true
+                // Ignore errors so that we can auto recover by retrying
+                unstable err.toString()
+            }
+        }
+    }
+}
+
+def runClient(isProducer, params) {
+    def envs = [
+        'JAVA_HOME': "/opt/jdk-13.0.1",
+        'M2_HOME': "/opt/apache-maven-3.6.3",
+        'AWS_KVS_LOG_LEVEL': params.AWS_KVS_LOG_LEVEL,
+        'CANARY_STREAM_NAME': "${env.JOB_NAME}",
+        'CANARY_LABEL': "${params.RUNNER_LABEL}",
+        'CANARY_TYPE': params.CANARY_TYPE,
+        'FRAGMENT_SIZE_IN_BYTES' : params.FRAGMENT_SIZE_IN_BYTES,
+        'CANARY_DURATION_IN_SECONDS': params.CANARY_DURATION_IN_SECONDS,
+        'AWS_DEFAULT_REGION': params.AWS_DEFAULT_REGION,
+    ].collect({k,v -> "${k}=${v}" })
+
+    // TODO: get the branch and version from orchestrator
+    if (params.FIRST_ITERATION) {
+
+        // TODO: Move to deletDir(). deleteDir() causes an exception right now 
+        sh """
+            cd $WORKSPACE
+            sudo rm -rf *
```

Comment:
why do we need sudo?

Reply:
Because the open-source directories have some files that cannot be removed without sudo. I know it is not the best, but this is the only way right now.

Reply:
Fixed after moving to CMAKE_INSTALL_PREFIX